airflow-core/src/airflow/jobs/triggerer_job_runner.py

```diff
@@ -603,8 +606,42 @@ def update_triggers(self, requested_trigger_ids: set[int]):
         adds them to the deques so the subprocess can actually mutate the running
         trigger set.
         """
+        from airflow.models.serialized_dag import SerializedDagModel
+
         render_log_fname = log_filename_template_renderer()
 
+        @provide_session
+        def create_workload(trigger: Trigger, session: Session = NEW_SESSION) -> workloads.RunTrigger:
+            if trigger.task_instance:
+                log_path = render_log_fname(ti=trigger.task_instance)
+                serialized_dag = SerializedDagModel.get_serialized_dag_by_version(
+                    dag_id=trigger.task_instance.dag_id,
+                    dag_version_id=trigger.task_instance.dag_version_id,
+                    session=session,
+                )
+                ser_ti = workloads.TaskInstance.model_validate(trigger.task_instance, from_attributes=True)
+
+                # When producing logs from TIs, include the job id producing the logs to disambiguate it.
+                self.logger_cache[new_id] = TriggerLoggingFactory(
+                    log_path=f"{log_path}.trigger.{self.job.id}.log",
+                    ti=ser_ti,  # type: ignore
+                )
+
+                return workloads.RunTrigger(
+                    classpath=trigger.classpath,
+                    id=new_id,
+                    encrypted_kwargs=trigger.encrypted_kwargs,
+                    ti=ser_ti,
+                    timeout_after=trigger.task_instance.trigger_timeout,
+                    dag=serialized_dag.data if serialized_dag else None,
+                )
+            return workloads.RunTrigger(
+                classpath=trigger.classpath,
+                id=new_id,
+                encrypted_kwargs=trigger.encrypted_kwargs,
+                ti=None,
+            )
```

Comment:
Non-blocking nitpick: it seems like you could consolidate these by creating a run_trigger_kwargs dict above, modifying what needs modifying in the `if trigger.task_instance` block, and calling `return workloads.RunTrigger(**run_trigger_kwargs) ` rather than having two different return paths which may eventually get out of sync if one gets changed, but that's likely just personal preference, feel free to just resolve this.